ui/pages/confirm-transaction-base/transaction-alerts/transaction-alerts.js

```diff
@@ -96,39 +95,33 @@ const TransactionAlerts = ({
           type="warning"
         />
       )}
-      {balanceError &&
-      chainId === MAINNET_CHAIN_ID &&
-      type === TRANSACTION_TYPES.DEPLOY_CONTRACT ? (
+      {balanceError && type === TRANSACTION_TYPES.DEPLOY_CONTRACT ? (
         <ActionableMessage
           className="actionable-message--warning"
           message={
-            <Typography variant={TYPOGRAPHY.H7} align="left">
-              {t('insufficientCurrency', [nativeCurrency, networkName])}{' '}
-              <Button
-                type="link"
-                className="transaction-alerts__link"
-                onClick={showBuyModal}
-              >
-                {t('buyEth')}
-              </Button>{' '}
-              {t('orDeposit')}
-            </Typography>
-          }
-          useIcon
-          iconFillColor="#d73a49"
-          type="danger"
-        />
-      ) : null}
-      {balanceError &&
-      chainId !== MAINNET_CHAIN_ID &&
-      type === TRANSACTION_TYPES.DEPLOY_CONTRACT ? (
-        <ActionableMessage
-          className="actionable-message--warning"
-          message={
-            <Typography variant={TYPOGRAPHY.H7} align="left">
-              {t('insufficientCurrency', [nativeCurrency, networkName])}
-              {t('buyOther', [nativeCurrency])}
-            </Typography>
+            isBuyableChain ? (
+              <Typography variant={TYPOGRAPHY.H7} align="left">
+                {t('insufficientCurrencyBuyOrDeposit', [
+                  nativeCurrency,
+                  networkName,
+                  <Button
+                    type="inline"
+                    className="confirm-page-container-content__link"
+                    onClick={showBuyModal}
+                    key={`${nativeCurrency}-buy-button`}
+                  >
+                    {t('buyAsset', [nativeCurrency])}
+                  </Button>,
+                ])}
+              </Typography>
+            ) : (
+              <Typography variant={TYPOGRAPHY.H7} align="left">
+                {t('insufficientCurrencyDeposit', [
+                  nativeCurrency,
+                  networkName,
+                ])}
+              </Typography>
+            )
           }
           useIcon
           iconFillColor="#d73a49"
```

Comment:
Can we update this one while we are at it?

```suggestion
          iconFillColor="var(--color-error-default)"
```